network-map-viewer/src/components/network/substation-layer.js

```diff
@@ -0,0 +1,220 @@
+                        );
+
+                    // sorted distinct nominal voltages for this substation
+                    const nominalVoltages = [
+                        ...new Set(
+                            substation.voltageLevels
+                                .map(
+                                    (voltageLevel) =>
+                                        voltageLevel.nominalVoltage
+                                )
+                                .sort(
+                                    (nominalVoltage1, nominalVoltage2) =>
+                                        nominalVoltage1 - nominalVoltage2
+                                )
+                        ),
+                    ];
+
+                    // add to global map of meta voltage levels indexed by nominal voltage
+                    Array.from(voltageLevelsByNominalVoltage.entries()).forEach(
+                        (e) => {
+                            const nominalVoltage = e[0];
+                            const voltageLevels = e[1];
+
+                            let metaVoltageLevels =
+                                metaVoltageLevelsByNominalVoltage.get(
+                                    nominalVoltage
+                                );
+                            if (!metaVoltageLevels) {
+                                metaVoltageLevels = [];
+                                metaVoltageLevelsByNominalVoltage.set(
+                                    nominalVoltage,
+                                    metaVoltageLevels
+                                );
+                            }
+                            metaVoltageLevels.push({
+                                voltageLevels,
+                                nominalVoltageIndex:
+                                    nominalVoltages.indexOf(nominalVoltage),
+                            });
+                        }
+                    );
+                });
+            }
+
+            // sort the map by descending nominal voltages
+            const metaVoltageLevelsByNominalVoltageArray = Array.from(
+                metaVoltageLevelsByNominalVoltage
+            )
+                .map((e) => {
+                    return { nominalVoltage: e[0], metaVoltageLevels: e[1] };
+                })
+                .sort((a, b) => b.nominalVoltage - a.nominalVoltage);
+
+            this.setState({
+                metaVoltageLevelsByNominalVoltage:
+                    metaVoltageLevelsByNominalVoltageArray,
+            });
+        }
+
+        if (
+            changeFlags.dataChanged ||
+            props.getNameOrId !== oldProps.getNameOrId ||
+            props.filteredNominalVoltages !== oldProps.filteredNominalVoltages
+        ) {
+            let substationsLabels = props.data;
+
+            if (
+                props.network != null &&
+                props.geoData != null &&
+                props.filteredNominalVoltages != null
+            ) {
```

Comment:
does null and undefined are two expected value ? if not, prefer short version 
  ```suggestion
                !props.network && !props.geoData && !props.filteredNominalVoltages
            ) {
```